jdaviz/core/template_mixin.py

```diff
@@ -1677,6 +1679,69 @@ def _check_dataset_spectral_subset_valid(self, event={}, return_ranges=False):
             return self.spectral_subset_valid
 
 
+class NonFiniteUncertaintyMismatchMixin(VuetifyTemplate, HubListener):
+    """ Adds a traitlet that identifies if there are any finite data values
+        that correspond to a non-finite uncertainty at that index.
+
+        In model fitting, the presence of finite, fittable data with corresponding
+        non-finite uncertainties can cause issues. Finite data values will be
+        filtered out in this case which may be undesirable. This traitlet when
+        True triggers a warning in the model fitting plugin (in specviz only,
+        currently) if there are any finite values with non-finite uncertainties.
+
+        Note that if a the uncertainty array is FULLY non-finite and the data is
+        FULLY finite, uncertainties will be set to None (in the Specviz parser),
+        so this traitlet will be False in that case (and therefore no warning
+        message displayed in the plugin).
+    """
+
+    non_finite_uncertainty_mismatch = Bool(False).tag(sync=True)
+
+    # every time a data/subset selection is changed, check the data selection and
+    # its uncertainties to see if there are any finite data elements with
+    # uncertainties. Warn in plugin if this occurs.
+
+    # note! the 'n/a' traitlet doesn't exist. we are watching it here because
+    # DatasetSpectralSubsetValidMixin is also watching the exact set
+    # of real traitlets, which creates a bug where only the last inhereted method
+    # will run, so we need to spoof it like this.
+    @observe("n/a", "dataset_selected", "spectral_subset_selected")
+    def _check_non_finite_uncertainty_mismatch(self, event={}):
+
+        if not hasattr(self, 'dataset') or self.dataset_selected == '':
+            # during initial init, this can trigger before the component is initialized
+            return
+        try:
+            spec = self._get_1d_spectrum()  # this is a method in model fitting
+        except AttributeError:
```

Comment:
technically an attribute error could be something else and raising this `ValueError` could confuse us in debugging in the future.  Maybe its safe to just have an `if not hasattr(self, '_get_1d_spectrum'): raise NotImplementedError("_get_1d_spectrum() must be available in this plugin to use NonFiniteUncertaintyMismatchMixin")` before trying to call `self._get_1d_spectrum()`.

Reply:
ok changed this to your suggestion